include/spblas/vendor/rocsparse/multiply_spgemm.hpp

```diff
@@ -0,0 +1,326 @@
+  }
+
+  auto result_shape() {
+    return this->result_shape_;
+  }
+
+  auto result_nnz() {
+    return this->result_nnz_;
+  }
+
+  template <matrix A, matrix B, matrix C, matrix D>
+    requires __detail::has_csr_base<A> && __detail::has_csr_base<B> &&
+             __detail::is_csr_view_v<C> && __detail::has_csr_base<D>
+  void multiply_compute(A&& a, B&& b, C&& c, D&& d) {
+    auto a_base = __detail::get_ultimate_base(a);
+    auto b_base = __detail::get_ultimate_base(b);
+    auto d_base = __detail::get_ultimate_base(d);
+    using matrix_type = decltype(a_base);
+    using input_type = decltype(b_base);
+    using output_type = std::remove_reference_t<decltype(c)>;
+    using value_type = typename matrix_type::scalar_type;
+
+    size_t buffer_size = 0;
+
+    auto alpha_optional = __detail::get_scaling_factor(a, b);
+    value_type alpha = alpha_optional.value_or(1);
+    auto beta_optional = __detail::get_scaling_factor(d);
+    value_type beta = beta_optional.value_or(1);
+    auto handle = this->handle_.get();
+    // Create sparse matrix A in CSR format
+    __rocsparse::throw_if_error(rocsparse_create_csr_descr(
+        &this->mat_a_, __backend::shape(a_base)[0], __backend::shape(a_base)[1],
+        a_base.values().size(), a_base.rowptr().data(), a_base.colind().data(),
+        a_base.values().data(),
+        to_rocsparse_indextype<typename matrix_type::offset_type>(),
+        to_rocsparse_indextype<typename matrix_type::index_type>(),
+        rocsparse_index_base_zero, to_rocsparse_datatype<value_type>()));
+    __rocsparse::throw_if_error(rocsparse_create_csr_descr(
+        &this->mat_b_, __backend::shape(b_base)[0], __backend::shape(b_base)[1],
+        b_base.values().size(), b_base.rowptr().data(), b_base.colind().data(),
+        b_base.values().data(),
+        to_rocsparse_indextype<typename input_type::offset_type>(),
+        to_rocsparse_indextype<typename input_type::index_type>(),
+        rocsparse_index_base_zero,
+        to_rocsparse_datatype<typename input_type::scalar_type>()));
+    __rocsparse::throw_if_error(rocsparse_create_csr_descr(
+        &this->mat_c_, __backend::shape(a_base)[0], __backend::shape(b_base)[1],
+        0, c.rowptr().data(), NULL, NULL,
+        to_rocsparse_indextype<typename output_type::offset_type>(),
+        to_rocsparse_indextype<typename output_type::index_type>(),
+        rocsparse_index_base_zero,
+        to_rocsparse_datatype<typename output_type::scalar_type>()));
+    __rocsparse::throw_if_error(rocsparse_create_csr_descr(
+        &this->mat_d_, __backend::shape(d_base)[0], __backend::shape(d_base)[1],
+        d_base.values().size(), d_base.rowptr().data(), d_base.colind().data(),
+        d_base.values().data(),
+        to_rocsparse_indextype<typename output_type::offset_type>(),
+        to_rocsparse_indextype<typename output_type::index_type>(),
+        rocsparse_index_base_zero,
+        to_rocsparse_datatype<typename output_type::scalar_type>()));
```

Comment:
maybe we could have something like:

using output_scalar_t = typename output_type::scalar_type;
It would simplify the writings.

Reply:
I have merged them into a helper function to create a matrix handle easily from the csr_view